src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -209,6 +215,16 @@ export interface D2ItemFactoryType {
   createItemIndex(item: DimItem): string;
 }
 
+export interface DimMasterwork {
+  progress: number | undefined;
```

Comment:
A shorthand for this is `progress?: number`.

Reply:
I accept but don't understand.  Is that a ternary operator?

Reply:
No, it specifies that the property is optional.